### ISSUE TITLE
Tuple selectors on calls of polymorphic functions don't typecheck correctly

### ISSUE THREAD

**Issue by RandomActsOfGrammar** [melt-umn/silver#486]
Consider the following code:
```
local lst::[(Integer, Integer)] = [(0, 1), (1, 2), (2, 3)];
local x::Integer = head(lst).1;
```
This would appear to be correct, but it fails with this error message:
```
error: Local x has type Integer but the expression being assigned to it has type (Integer, Integer)
```
Suppose we change the code to the following:
```
local lst::[(Integer, Integer)] = [(0, 1), (1, 2), (2, 3)];
local h::(Integer, Integer) = head(lst);
local x::Integer = h.1;
```
This does not have an error.

Suppose we have the following function:
```
function id
a ::= a::a
{
  return a;
}
```
Then `local y::Integer = id((1, 3)).1;` will give us the same type of error as above.  If, however, we change the `id` function to operate on pairs specifically, like so
```
function id
(a, b) ::= a::(a, b)
{
  return a;
}
```
the same definition for `y` does not give us an error.

If we change the selector to `.2` instead of `.1` on the examples which draw the error, we get `error: Invalid tuple selector index.`

Given that this only happens with polymorphic functions, I would guess it has something to do with type substitutions, but I'm not sure.

**Comment by krame505** [comment]
Looks like we should probably be applying `tuple.upSubst` to `tuple.typerep` before computing `len` in the `selector` production.  @alexandra-hanson do you want to take a shot at fixing this?

**Comment by RandomActsOfGrammar** [comment]
This was fixed by #496, so I'm closing it.

@alexandra-hanson For future reference, you should include something like "Fixes #<bug number>", "Closes #<bug number>", or "Resolves #<bug number>" in the description of your pull requests.  This will automatically close the issue you are fixing when the pull request is merged.

**Comment by alexandra-hanson** [comment]
@RandomActsOfGrammar Thanks! Sorry about that. I thought just mentioning the issue would close it -- I'll be sure to use the appropriate keywords from now.
